api/metric/api.go

```diff
@@ -234,3 +243,16 @@ type resourceOption resource.Resource
 func (r resourceOption) Apply(config *Config) {
 	config.Resource = resource.Resource(r)
 }
+
+// WithLibraryName applies provided library name.
```

Comment:
To me, it looks like an option that libraries and applications basically shouldn't use, because the library name should be provided by the meter provider. Maybe document the option to be mostly for meter provider implementations?

Reply:
Done.